addons/website/static/src/js/http_cookie.js

```diff
@@ -0,0 +1,18 @@
+/** @odoo-module **/
+
+import cookieUtils from 'web.utils.cookies';
+
+const originFunc = cookieUtils.checkCookie;
+cookieUtils.checkCookie = (type, name) => {
+    const result = originFunc.apply(cookieUtils, [type, name]);
+    if (result && type === 'optional') {
+        const consents = JSON.parse(cookieUtils.get_cookie('accepted_cookie_types') || '{}');
+        if ('optional' in consents) {
+            return consents['optional'];
+        }
+        return !document.getElementById('cookies-consent-essential');
```

Comment:
Also make this return first in the `ìf()` ?
Won't have to parse the cookies, if the cookies bar is disabled, no need to do all that -> Return true
```js
        if (!document.getElementById('cookies-consent-essential')) {
            // Cookies bar is disabled on this website
            return true;
        }
        const consents = JSON.parse(cookieUtils.get_cookie('accepted_cookie_types') || '{}');
        if ('optional' in consents) {
            return consents['optional'];
        }
```

Reply:
Could you apply the same in the python def? (no need to check and parse cookies if cookies bar disabled)
```py
    @classmethod
    def _is_allowed_cookie(cls, cookie_type, cookie_name='undefined'):
        result = super()._is_allowed_cookie(cookie_type, cookie_name)
        if result and cookie_type == 'optional':
            accepted_cookie_types = json_scriptsafe.loads(request.httprequest.cookies.get('accepted_cookie_types', '{}'))
            if 'optional' in accepted_cookie_types:
                return accepted_cookie_types['optional']
            return not request.env['website'].get_current_website().cookies_bar

        # Pass-through if already forbidden for another reason or a type that
        # is not restricted by the website module.
        return result
```

```py
    @classmethod
    def _is_allowed_cookie(cls, cookie_type, cookie_name='undefined'):
        result = super()._is_allowed_cookie(cookie_type, cookie_name)
        if result and cookie_type == 'optional':
            if not request.env['website'].get_current_website().cookies_bar:
                # Cookies bar is disabled on this website
                return True
            accepted_cookie_types = json_scriptsafe.loads(request.httprequest.cookies.get('accepted_cookie_types', '{}'))
            if 'optional' in accepted_cookie_types:
                return accepted_cookie_types['optional']

        # Pass-through if already forbidden for another reason or a type that
        # is not restricted by the website module.
        return result
```

Reply:
Applied change and added a `return False` at the end of the first `if` block.